lib/Transforms/Scalar/LoopUnrollPass.cpp

```diff
@@ -101,13 +101,13 @@ namespace {
   public:
     static char ID; // Pass ID, replacement for typeid
     LoopUnroll(int T = -1, int C = -1, int P = -1, int R = -1) : LoopPass(ID) {
-      CurrentThreshold = (T == -1) ? UnrollThreshold : unsigned(T);
+      CurrentThreshold = (T == -1) ? UnrollThreshold : NullOpt<unsigned>(unsigned(T));
```

Comment:
I have also run into this compilation error. The issue is ambiguity of type for the two sides of the ternary operator, as you have pointed out.
So, there is also another way to fix it:
```cpp
CurrentThreshold = (T == -1) ? unsigned(UnrollThreshold) : unsigned(T);
```
Given that the LHS of the assignment (CurrentThreshold) is an unsigned int (defined on line 139), this way seems simpler?

Reply:
Also applies to the next 3 changes.

Reply:
Makes sense. I guess I misread this. Somehow I remember thinking the LHS was NullOpt